src/auth/pushpadHelper.js

```diff
@@ -8,62 +8,70 @@ export const initPushpad = (uid, username, token) => {
     return;
   }
 
-  fetch(`${BUSYPUSH_ENDPOINT}/api/getSignature`, {
-    method: 'POST',
-    headers: {
-      'Accept': 'application/json, text/plain, */*',
-      'Content-Type': 'application/json'
-    },
-    body: JSON.stringify({
-      uid,
-      username,
-      token,
-    }),
-  })
-    .then(jsonRes => jsonRes.json())
-    .then((res) => {
-      if (!(res && res.isValid)) {
-        return;
-      }
+  // code is generated by pushpad
+  /* eslint-disable */
+  (function(p,u,s,h,x) {
+    p.pushpad = p.pushpad || function() {
+      (p.pushpad.q = p.pushpad.q || []).push(arguments)
+    };
+    h = u.getElementsByTagName('head')[0];
+    x = u.createElement('script');
+    x.async = 1;
+    x.src = s;
+    h.appendChild(x);
+  })(window,document, 'https://pushpad.xyz/pushpad.js');
 
-      const { signatureId } = res;
+  pushpad('status', (isSubscribed) => {
+    if (isSubscribed) {
+      return;
+    }
 
-      // code is generated by pushpad
-      /* eslint-disable */
-      (function(p,u,s,h,x) {
-        p.pushpad = p.pushpad || function() {
-          (p.pushpad.q = p.pushpad.q || []).push(arguments)
-        };
-        h = u.getElementsByTagName('head')[0];
-        x = u.createElement('script');
-        x.async = 1;
-        x.src = s;
-        h.appendChild(x);
-      })(window,document, 'https://pushpad.xyz/pushpad.js');
-
-      pushpad('init', PUSHPAD_PROJECT_ID);
-      pushpad('uid', `${uid}`, signatureId);
-      pushpad('subscribe', (isSubscribed) => {
-        if (!isSubscribed) {
+    fetch(`${BUSYPUSH_ENDPOINT}/api/getSignature`, {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json, text/plain, */*',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        uid,
+        username,
+        token,
+      }),
+    })
+      .then(jsonRes => jsonRes.json())
+      .then((res) => {
+        if (!(res && res.isValid)) {
           return;
         }
 
-        // subscribe username in busy-push to check for its activities in Steem blockchain
-        fetch(`${BUSYPUSH_ENDPOINT}/api/subscribe`, {
-          method: 'POST',
-          headers: {
-            'Accept': 'application/json, text/plain, */*',
-            'Content-Type': 'application/json'
-          },
-          body: JSON.stringify({
-            uid,
-            username,
-            token,
-          }),
-        })
+        const { signatureId } = res;
+
+        pushpad('init', PUSHPAD_PROJECT_ID);
+        pushpad('uid', `${uid}`, signatureId);
+        pushpad('subscribe', (isSubscribed) => {
+          if (!isSubscribed) {
```

Comment:
I understand what you mean but it doesn't work like that. signature is fetched if user is not subscribed so after that pushpad will ask user to accept the permission, if user accept this time, the callback will return true so the code will be executed in line 57.